internal-dns/src/names.rs

```diff
@@ -6,6 +6,13 @@
 
 use omicron_uuid_kinds::{OmicronZoneUuid, SledUuid};
 
+/// Name for the special boundary NTP DNS name
+///
+/// chrony does not support SRV records. This name resolves to AAAA records for
+/// each boundary NTP zone, and then we can point internal NTP chrony instances
+/// at this name for it to find the boundary NTP zones.
+pub const BOUNDARY_NTP_DNS_NAME: &str = "boundary.ntp";
```

Comment:
My inclination would be to just make this `boundary-ntp` or something.  I'm not sure what the extra level of hierarchy would get us (and it seems a little weird because `ntp` is not also a name or zone that we're using).